lib/Storage/LocalStorageClient.cpp

```diff
@@ -0,0 +1,133 @@
+/// \file LocalStorageClient.cpp
+/// \brief Implementation of StorageClient operations on the local filesystem
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Storage/LocalStorageClient.h"
+#include "revng/Support/PathList.h"
+
+namespace revng::detail {
+
+class LocalReadableFile : public ReadableFile {
+private:
+  std::unique_ptr<llvm::MemoryBuffer> Buffer;
+
+public:
+  LocalReadableFile(std::unique_ptr<llvm::MemoryBuffer> &&Buffer) :
+    Buffer(std::move(Buffer)) {}
+  ~LocalReadableFile() override = default;
+
+  llvm::MemoryBuffer &buffer() override { return *Buffer; };
+};
+
+class LocalWritableFile : public WritableFile {
+private:
+  std::unique_ptr<llvm::raw_fd_ostream> OS;
+
+public:
+  LocalWritableFile(std::unique_ptr<llvm::raw_fd_ostream> &&OS) :
+    OS(std::move(OS)) {}
+  ~LocalWritableFile() override = default;
+  llvm::raw_pwrite_stream &os() override { return *OS; }
+  llvm::Error commit() override { return llvm::Error::success(); }
+};
+
+std::string LocalStorageClient::resolvePath(const llvm::StringRef Path) {
```

Comment:
Please review all the `StorageClient`s to prevent path traversals.

A `StorageClient` should act as a sort of sandbox. It should not be possible to write out of `Root` using it.

Reply:
Added a `normalizePath` function that removes the possibility of path traversal altogether